commands/core/site_install.drush.inc

```diff
@@ -55,7 +56,8 @@ function site_install_drush_command() {
         'example-value' => 'directory_name',
       ),
       'writable' => 'Make CMI and other dirs writable by both web and CLI users. Suitable for non Prod environments.',
-      'keep-config' => 'Keep CMI directories untouched. This preserves existing configuration.'
+      'keep-config' => 'Keep CMI directories untouched. This preserves existing configuration.',
```

Comment:
options that are version specific should get altered in so they aren't valid or shown for other versions. see user_drush_help_alter() for example.